Add, move, and tidy up scaladoc variables around collections

chrisphelps opened the pull request:
Partially fixes scala/collection-strawman#510:
- fixes doc generation errors in IterableOnceOps (and several others)
- does not address differences in IterableOps vs IterableOnceOps (future PR)

Mostly fixes scala/bug#10969:
- Addresses unbound references in the listed files and others
- Does not address any issues caused by the referenced bug scala/bug#9785

A couple questions I'd like some specific feedback on:
- StringOps defines a `$coll` variable but does not use that very much. Is it worth expanding the use of this variable or removing it entirely? This class does not rely much on inherited docs, nor are these docs inherited elsewhere.
- In StringOps, I adjusted all the docs to consistently capitalize `String`. Should ArrayOps do a similar thing for `Array`?

Commit:
Add, move, and tidy up scaladoc variables around collections

szeiger commented:
I suggest removing `$coll` and similar variables from both, `StringOps` and `ArrayOps`. Now that these are stand-alone value classes which don't inherit any method, these definitions don't serve any useful purpose.

Both "array" and "string" can be used formally for the `Array` and `String` types or informally, which should be the same in the `ArrayOps` and `StringOps` docs, so I'd go for consistency.

szeiger commented on src/library/scala/collection/LazyZipOps.scala:
```diff
@@ -2,7 +2,13 @@ package scala.collection
+/** Decorator representing lazily zipped pairs.
+  *
+  * @define coll pair
+  * @define willNotTerminateInf
```
Both appear to be unused at the moment. We will need `willNotTerminateInf` but I think `coll` is unnecessary in these LazyZip classes (because they don't extend any standard collection traits, just like `ArrayOps` and `StringOps`)

chrisphelps replied:
The `coll` here is used for the `IterableOnce` methods which are included via implicit conversion to `View`. Without these, we get "view" in the docs, but also a number of extra warnings when building the docs.

szeiger commented on src/library/scala/collection/StringOps.scala:
```diff
@@ -510,14 +513,14 @@ final class StringOps(private val s: String) extends AnyVal {
   @inline final def addString(b: StringBuilder): StringBuilder = b.append(s)
 
-  /** Appends this string to a string builder using a separator string. */
+  /** Appends this String to a String Builder using a separator String. */
```
I'd go with `StringBuilder` if it's capitalized

chrisphelps commented:
Thanks for the feedback @szeiger. I'll update in the next few days.

szeiger commented:
@chrisphelps Any update on this?

chrisphelps commented:
I removed the `$coll` from `StringOps` and `ArrayOps`, and converted all usages of `String` to the informal use across `StringOps`. The `coll` definitions were left in `LazyZip` as those are used by methods from `IterableOnce`.
